src/node_crypto.cc

```diff
@@ -4038,12 +4071,60 @@ void Sign::SignUpdate(const FunctionCallbackInfo<Value>& args) {
   sign->CheckThrow(err);
 }
 
+static int Node_SignFinal(EVP_MD_CTX *mdctx, unsigned char *md, unsigned int *s,
+                          EVP_PKEY *pkey, int padding, int pss_saltlen) {
+  unsigned char m[EVP_MAX_MD_SIZE];
+  unsigned int m_len;
+  int i = 0;
+  EVP_PKEY_CTX *pkctx = nullptr;
+
+  *s = 0;
+  if (!EVP_DigestFinal_ex(mdctx, m, &m_len))
+    return i;
+
+  if (mdctx->digest->flags & EVP_MD_FLAG_PKEY_METHOD_SIGNATURE) {
+    size_t sltmp = static_cast<size_t>(EVP_PKEY_size(pkey));
```

Comment:
It means "signature length", `Node_SignFinal` is mostly based on `EVP_SignFinal` with some adjustments. To be honest, I do not fully understand its relation to the key size, it appears to be related to `EVP_PKEY_FLAG_AUTOARGLEN`.